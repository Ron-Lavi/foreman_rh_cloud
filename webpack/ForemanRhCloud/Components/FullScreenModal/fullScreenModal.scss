.modal-dialog {
  width: 95%;
  height: 95%;
  margin: 30px 0 0 30px;

  .modal-body {
    padding: 0;
    display: flex;
    height: -webkit-fill-available;
    height: -moz-available;
    height: fill-available;

    .terminal {
      margin: 0 -20px;
      height: 100%;
    }
  }
}
