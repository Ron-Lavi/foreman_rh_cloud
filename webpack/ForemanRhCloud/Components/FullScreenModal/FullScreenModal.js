import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Icon, noop } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import Terminal from '../Terminal';
import './fullScreenModal.scss';

const FullScreenModal = ({
  showFullScreen,
  toggleFullScreen,
  terminalProps,
}) => (
  <Modal show={showFullScreen} onHide={toggleFullScreen}>
    <Modal.Header>
      <button
        className="close"
        onClick={toggleFullScreen}
        aria-hidden="true"
        aria-label="Close"
      >
        <Icon type="pf" name="close" />
      </button>
      <Modal.Title>{__('Full Screen')}</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <Terminal autoScroll={false} {...terminalProps} />
    </Modal.Body>
  </Modal>
);

FullScreenModal.propTypes = {
  showFullScreen: PropTypes.bool,
  toggleFullScreen: PropTypes.func,
  terminalProps: PropTypes.shape({
    exitCode: PropTypes.string,
    logs: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.string),
      PropTypes.string,
    ]),
    error: PropTypes.string,
  }),
};

FullScreenModal.defaultProps = {
  showFullScreen: false,
  toggleFullScreen: noop,
  terminalProps: {},
};

export default FullScreenModal;
