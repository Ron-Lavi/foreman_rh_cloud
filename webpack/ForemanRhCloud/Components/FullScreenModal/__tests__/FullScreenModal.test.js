import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import FullScreenModal from '../FullScreenModal';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('FullScreenModal', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(FullScreenModal, fixtures));
});
