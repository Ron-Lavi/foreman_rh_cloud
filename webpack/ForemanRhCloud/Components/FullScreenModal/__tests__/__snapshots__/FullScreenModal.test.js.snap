// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenModal rendering render without Props 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={false}
    closeLabel="Close"
  >
    <button
      aria-hidden="true"
      aria-label="Close"
      className="close"
      onClick={[Function]}
    >
      <Icon
        name="close"
        type="pf"
      />
    </button>
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Full Screen
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <Terminal
      autoScroll={false}
      error={null}
      exitCode=""
      logs={null}
    />
  </ModalBody>
</Modal>
`;
